contrib/IECoreAlembic/src/IECoreAlembic/AlembicScene.cpp

```diff
@@ -875,6 +1424,262 @@ class AlembicScene::AlembicWriter : public AlembicIO
+				{
+					setProperty<ON3fProperty>( name, time, data );
+				}
+			}
+			else if( const IECore::V3dData *data = runTimeCast<const IECore::V3dData>( attribute ) )
+			{
+				if( data->getInterpretation() == GeometricData::Interpretation::Vector )
+				{
+					setProperty<OV3dProperty>( name, time, data );
+				}
+				else if( data->getInterpretation() == GeometricData::Interpretation::Point )
+				{
+					setProperty<OP3dProperty>( name, time, data );
+				}
+				else if( data->getInterpretation() == GeometricData::Interpretation::Normal )
+				{
+					setProperty<ON3dProperty>( name, time, data );
+				}
+			}
+			else if( const IECore::Box2iData *data = runTimeCast<const IECore::Box2iData>( attribute ) )
+			{
+				setProperty<OBox2iProperty>( name, time, data );
+			}
+			else if( const IECore::Box2fData *data = runTimeCast<const IECore::Box2fData>( attribute ) )
+			{
+				setProperty<OBox2fProperty>( name, time, data );
+			}
+			else if( const IECore::Box2dData *data = runTimeCast<const IECore::Box2dData>( attribute ) )
+			{
+				setProperty<OBox2dProperty>( name, time, data );
+			}
+			else if( const IECore::Box3iData *data = runTimeCast<const IECore::Box3iData>( attribute ) )
+			{
+				setProperty<OBox3iProperty>( name, time, data );
+			}
+			else if( const IECore::Box3fData *data = runTimeCast<const IECore::Box3fData>( attribute ) )
+			{
+				setProperty<OBox3fProperty>( name, time, data );
+			}
+			else if( const IECore::Box3dData *data = runTimeCast<const IECore::Box3dData>( attribute ) )
+			{
+				setProperty<OBox3dProperty>( name, time, data );
+			}
+			else if( const IECore::M33fData *data = runTimeCast<const IECore::M33fData>( attribute ) )
+			{
+				setProperty<OM33fProperty>( name, time, data );
+			}
+			else if( const IECore::M33dData *data = runTimeCast<const IECore::M33dData>( attribute ) )
+			{
+				setProperty<OM33dProperty>( name, time, data );
+			}
+			else if( const IECore::M44fData *data = runTimeCast<const IECore::M44fData>( attribute ) )
+			{
+				setProperty<OM44fProperty>( name, time, data );
+			}
+			else if( const IECore::M44dData *data = runTimeCast<const IECore::M44dData>( attribute ) )
+			{
+				setProperty<OM44dProperty>( name, time, data );
+			}
+			else if( const IECore::QuatfData *data = runTimeCast<const IECore::QuatfData>( attribute ) )
+			{
+				setProperty<OQuatfProperty>( name, time, data );
+			}
+			else if( const IECore::QuatdData *data = runTimeCast<const IECore::QuatdData>( attribute ) )
+			{
+				setProperty<OQuatdProperty>( name, time, data );
+			}
+			else if( const IECore::Color3fData *data = runTimeCast<const IECore::Color3fData>( attribute ) )
+			{
+				setProperty<OC3fProperty>( name, time, data );
+			}
+			else if( const IECore::Color4fData *data = runTimeCast<const IECore::Color4fData>( attribute ) )
+			{
+				setProperty<OC4fProperty>( name, time, data );
+			}
+			else
+			{
+				throw IECore::Exception(
```

Comment:
Throwing here will prevent us from using the SceneWriter with any scene that has shader assignments. I suggest we just make it a warning message instead...